test/unit/eigensolver/test_eigensolver.cpp

```diff
@@ -114,7 +115,7 @@ void testEigensolverCorrectness(const blas::Uplo uplo, Matrix<const T, Device::C
   CHECK_MATRIX_NEAR(res, workspace, 2 * m * TypeUtilities<T>::error, 2 * m * TypeUtilities<T>::error);
 }
 
-template <class T, Backend B, Device D>
+template <class T, Backend B, Device D, bool preallocated = false>
```

Comment:
Since this is a test I would just remove the default and always explicitly say whether or not the result should be preallocated. Possibly with tag types/enums so that one doesn't have to wonder at the callsite if the boolean is `preallocated` or `allocated` (or whatever one would call the opposite).

Reply:
I used `enum class` (much cleaner, thanks!) but then I wanted to check the `else` statement (since it's no longer a boolean). I added a `static_assert` in the `else` branch, but it's not clean at all... Should I just drop it, since the `enum class` should only contain valid values?

Reply:
Yep, this looks reasonable enough to me. It's quite unlikely that enum is going to be extended anyway (but you never know). No need to change anything from my point of view, but the `Allocation` template parameter need not be a template parameter either, and then you could just do a plain old `switch-case`. But as I said, I'm happy with the current version as well.